src/index.js

```diff
@@ -26,44 +42,40 @@ const clearField = () => {
 form.addEventListener('submit', () => {
   if (inputField.value === '') return;
 
-  // Access Template Elements
-  const template = document.querySelector('#list-item-template');
-  const templateClone = template.content.cloneNode(true);
-  const taskContent = templateClone.querySelector('[data-list-item-text]');
-  const checkBox = templateClone.querySelector('[data-list-item-checkbox]');
-
-  // Create a new Todo
   const todoTemplate = new Todo(todoList.length, inputField.value, false);
   todoList.push(todoTemplate);
-  saveList();
-  taskContent.innerText = inputField.value;
-  if (checkBox.checked) {
-    taskContent.classList.add('.line-through');
-    checkBox.checked = true;
-  } else {
-    taskContent.classList.remove('.line-through');
-    checkBox.checked = false;
-  }
-  todoListContainer.appendChild(templateClone);
+  renderTodo(todoTemplate);
+  localStorage.setItem(TODO_LIST_KEY, JSON.stringify(todoList));
   clearField();
 });
 
-todoList.forEach((todo) => {
-  const template = document.querySelector('#list-item-template');
-  const templateClone = template.content.cloneNode(true);
-  const taskContent = templateClone.querySelector('[data-list-item-text]');
-  const checkBox = templateClone.querySelector('[data-list-item-checkbox]');
-  checkBox.addEventListener('change', () => {
-    // Save checkbox event in local storage
-    checkBoxEvent(todo, checkBox, taskContent, saveList);
-  });
-  taskContent.innerText = todo.description;
-  checkBox.checked = todo.completed;
-  saveList();
-  if (checkBox.checked) {
-    taskContent.classList.add('line-through');
-  } else {
-    taskContent.classList.remove('line-through');
+todoListContainer.addEventListener('click', (e) => {
+  if (!e.target.matches('[data-button-delete]')) return;
+
+  const parent = e.target.closest('.list-item');
+  const todoIndex = parseInt(parent.dataset.todoIndex);
+  parent.remove(); // removes from the screen
+  todoList = todoList.filter((todo) => todo.index !== todoIndex); // removes from the list
+  localStorage.setItem(TODO_LIST_KEY, JSON.stringify(todoList));
+});
+
+clearAllCompleted.addEventListener('click', () => {
+  todoList = todoList.filter((todo) => todo.completed === false);
+  // saveList();
+  localStorage.setItem(TODO_LIST_KEY, JSON.stringify(todoList));
+  location.reload();
+});
+
```

Comment:
- [ ] The issue pointed out with the previous code reviewer still persists, kindly address it before you request another code review.
=> A bit of explanation about it, it's related to this hard requirement of this project: ` the property index set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5)` So, this mean, when you display values of indices, kindly don't start from zero as if you are displaying indices of an array, you are just displaying numbering of a list of tasks in more human readable style, so kindly make sure that the numbering starts counting from one not from zero.

- [ ] [Optional] The other hard requirement which should be mentioned on the previous review but wasn't (sorry about that) is: `Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.`
- [ ] ![Screenshot from 2021-08-21 19-40-47](https://user-images.githubusercontent.com/56886064/130330715-c7bd5b0a-9c97-45d3-abde-bf4dbb4ce408.png)
As You can see on the screenshot, the indices are not re-ordered after I removed some tasks

=> _As a note: I am marking this requirement as optional because we forgot to remind it to you on the previous review, but it should have been taken care of as a hard requirement_

Reply:
# Here's proof that the index issue has been resolved and the project is working absolutely **FINE**. I already lost 2 attempts just because of issue.

![Screenshot from 2021-08-21 23-00-29](https://user-images.githubusercontent.com/83048208/130331038-88764bdf-4361-4730-9168-31eba003632e.png)